component/org.wso2.carbon.identity.oauth2.grant.token.exchange/src/main/java/org/wso2/carbon/identity/oauth2/grant/token/exchange/utils/TokenExchangeUtils.java

```diff
@@ -858,46 +898,101 @@ private static boolean validateUsingCertificate(SignedJWT signedJWT, IdentityPro
         return signedJWT.verify(verifier);
     }
 
+    /**
+     * Method to resolve and return the matching local user account based on the provided claims.
+     * @param tokReqMsgCtx  OauthTokenReqMessageContext
+     * @param claims  Lookup claims
+     * @return  Matching local user account
+     * @throws IdentityOAuth2Exception  Error when resolving local user account
+     */
     public static User getLocalUser(OAuthTokenReqMessageContext tokReqMsgCtx,
-                                    String subjectIdentifier) throws IdentityOAuth2Exception {
+                                    Map<String, String> claims) throws IdentityOAuth2Exception {
```

Comment:
is this method added by you for this feature? If not, we may not be able to update this method signature as this is a public method.

Reply:
Yes, this was added by me in the previous pr